cpp/src/arrow/compute/kernels/hash_test.cc

```diff
@@ -417,6 +417,101 @@ TEST_F(TestHashKernel, BinaryResizeTable) {
                                            indices);
 }
 
+TEST_F(TestHashKernel, UniqueLargeBinary) {
```

Comment:
It looks like this is mostly pasting existing code. Instead, can you define a parameterized test class for binary hash kernel tests?